BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -5,40 +5,60 @@ public struct AttributedLabel: Element {
 
     public var attributedText: NSAttributedString
     public var numberOfLines: Int = 0
-    /// The scale to which pixel measurements will be rounded. Defaults to `UIScreen.main.scale`.
-    public var roundingScale: CGFloat = UIScreen.main.scale
+    public var isAccessibilityElement = false
+
+    /// An offset that will be applied to the rect used by `drawText(in:)`.
+    ///
+    /// This can be used to adjust the positioning of text within each line's frame, such as adjusting
+    /// the way text is distributed within the line height.
+    public var textRectOffset: CGPoint = .zero
 
     public init(attributedText: NSAttributedString) {
         self.attributedText = attributedText
     }
 
     public var content: ElementContent {
         struct Measurer: Measurable {
+            private static let prototypeLabel = LabelView()
 
-            var attributedText: NSAttributedString
-            var roundingScale: CGFloat
+            var model: AttributedLabel
 
             func measure(in constraint: SizeConstraint) -> CGSize {
-                var size = attributedText.boundingRect(
-                    with: constraint.maximum,
-                    options: [.usesLineFragmentOrigin],
-                    context: nil)
-                    .size
-                size.width = size.width.rounded(.up, by: roundingScale)
-                size.height = size.height.rounded(.up, by: roundingScale)
-
-                return size
+                let label = Self.prototypeLabel
+                model.update(label: label)
+                return label.sizeThatFits(constraint.maximum)
             }
         }
 
-        return ElementContent(measurable: Measurer(attributedText: attributedText, roundingScale: roundingScale))
+        return ElementContent(measurable: Measurer(model: self))
+    }
+
+    private func update(label: LabelView) {
+        label.attributedText = attributedText
+        label.numberOfLines = numberOfLines
+        label.isAccessibilityElement = isAccessibilityElement
+        label.textRectOffset = textRectOffset
     }
 
     public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
-        return UILabel.describe { (config) in
-            config[\.attributedText] = attributedText
-            config[\.numberOfLines] = numberOfLines
+        return LabelView.describe { (config) in
+            config.apply(update)
         }
     }
+}
 
+extension AttributedLabel {
+    private final class LabelView: UILabel {
```

Comment:
Private type here means I couldn't use `UIViewElement`.

Reply:
me rn

![tumblr_inline_n23y8zrALB1rz15mt](https://user-images.githubusercontent.com/327847/86675460-841e1480-bfae-11ea-9e50-4ca82c83e2d8.gif)

Reply:
(Though, wondering: Do we want to make this public so we don't need to re-implement this behaviour in MarketUI for other UIView-constructed views which need this same behaviour?)

Reply:
Is there an attribute to make it importable without being public/automatically imported? Like, how does `import UIKit.UIGestureRecognizerSubclass` work?

Reply:
This would be a custom module map, which is definitely doable.

Reply:
```
spec.module_map = 'source/module.modulemap'
```

And then in the module map do something like this: https://git.sqcorp.co/projects/IOS/repos/register/browse/Frameworks/RegisterUI/RegisterUI.modulemap

Reply:
Doesn't seem worth the effort for a 12-line class to me. I'd rather copy it.

Reply:
I guess, but the module also isn't much more work, and it would be nice to keep these in sync. Oh well, too bad the swift SPI stuff isnt ready for primetime yet.

Reply:
Why wouldn’t something like MarketLabel be built on top of AttributedLabel rather than the underlying view?

Reply:
This is more referring to using the underlying Label view for the controls we have to build with UIKit because they're highly stateful, eg the floating labels in text fields.